### ISSUE TITLE
format_timedelta throws a KeyError for the Esperanto locale

### ISSUE THREAD

**Issue by oakes** [python-babel/babel#107]
With version 1.3, I have the following code:

    format_timedelta(dt - utcnow(), add_direction=True, locale=loc)

When `loc` is "eo", I get the following error:

    Traceback (most recent call last):                                                               
      File "/<snip>/www.gittip.com/env/local/lib/python2.7/site-packages/algorithm.py", line 288, in run
        new_state = function(**deps.as_kwargs)                                                       
      File "/<snip>/www.gittip.com/env/local/lib/python2.7/site-packages/aspen/algorithms/website.py", line 75, in get_response_for_resource
        return {'response': resource.respond(request)}                                               
      File "/<snip>/www.gittip.com/env/local/lib/python2.7/site-packages/aspen/resources/dynamic_resource.py", line 56, in respond
        exec self.pages[1] in context                                                                
      File "/<snip>/www.gittip.com/www/%username/history/index.html.spt", line 15, in <module>       
        JOINED = _("You joined {0}.").format(to_age(participant.claimed_time))                       
      File "/<snip>/www.gittip.com/gittip/utils/i18n.py", line 53, in _to_age                        
        return to_age(delta, loc)                                                                    
      File "/<snip>/www.gittip.com/gittip/utils/i18n.py", line 14, in to_age                         
        return format_timedelta(dt - utcnow(), add_direction=True, locale=loc)                       
      File "/<snip>/www.gittip.com/env/local/lib/python2.7/site-packages/babel/dates.py", line 784, in format_timedelta
        pattern = patterns[plural_form]                                                              
      File "/<snip>/www.gittip.com/env/local/lib/python2.7/site-packages/babel/localedata.py", line 189, in __getitem__
        orig = val = self._data[key]                                                                 
    KeyError: 'one'

**Comment by etanol** [comment]
I cannot reproduce it, so it has probably been indirectly fixed.  Please reopen if necessary.

**Comment by jun66j5** [comment]
Reproduced with Babel 1.3 and 2.1.1.
```
Python 2.7.3 (default, Jun 22 2015, 19:33:41)
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import timedelta
>>> import babel
>>> babel.__version__
'2.1.1'
>>> from babel.core import Locale
>>> from babel.dates import format_timedelta
>>> eo = Locale.parse('eo')
>>> format_timedelta(timedelta(seconds=1), add_direction=True, locale=eo)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/dev/shm/babel-2.1.1/local/lib/python2.7/site-packages/babel/dates.py", line 784, in format_timedelta
    pattern = patterns[plural_form]
  File "/dev/shm/babel-2.1.1/local/lib/python2.7/site-packages/babel/localedata.py", line 189, in __getitem__
    orig = val = self._data[key]
KeyError: 'one'
```

The 168 locales have the same issue with Babel 2.1.1.
```
>>> from datetime import timedelta
>>> from babel.core import Locale
>>> from babel.dates import format_timedelta
>>> def try_format(locale):
...   try:
...     format_timedelta(timedelta(seconds=1), add_direction=True, locale=locale)
...     return True
...   except Exception as e:
...     print('%s => %s: %s' % (locale, e.__class__.__name__, e))
...     return False
...
>>> from babel.localedata import locale_identifiers
>>> failed = 0
>>> for locale in sorted(locale_identifiers()):
...   if not try_format(locale):
...     failed += 1
...
ak => KeyError: 'one'
ak_GH => KeyError: 'one'
asa => KeyError: 'one'
asa_TZ => KeyError: 'one'
be => KeyError: 'one'
be_BY => KeyError: 'one'
bem => KeyError: 'one'
bem_ZM => KeyError: 'one'
...
ve_ZA => KeyError: 'one'
vo => KeyError: 'one'
vun => KeyError: 'one'
vun_TZ => KeyError: 'one'
xh => KeyError: 'one'
xh_ZA => KeyError: 'one'
xog => KeyError: 'one'
xog_UG => KeyError: 'one'
>>> failed
168
```

**Comment by etanol** [comment]
Right.  There seems to be a problem with the import script for CLDR 23 in this case.  However, @sils1297 I don't know if we will do more 2.1.y releases or we'd better focus on a near 2.2.

This problem is fixed in both `master` and `2.x-maintenance` (soon to be renamed `2.2`, I guess).

**Comment by sils1297** [comment]
@etanol thanks for the clarification. This will be fixed in the 2.2 release then which we will do based on demand or if changes add up after some time simply.

**Comment by sils1297** [comment]
@oakes @jun66j5 if there is an active need to get the bugfix released ASAP please say so, then we'll of course backport it and do a release tomorrowish, other than that I'd avoid this because we're currently trying to lower maintenance efforts.
